float64 speed_ratio
---
bool success